samples/nrf9160/lwm2m_client/overlay-nbiot.conf

```diff
@@ -0,0 +1,6 @@
+CONFIG_LTE_NETWORK_MODE_NBIOT=y
+CONFIG_COAP_INIT_ACK_TIMEOUT_MS=15000
+CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME=90
```

Comment:
why do we need these overrides?

Reply:
When testing the sample in NB-IoT network, I've observed that 10s ACK timeout is still not enough sometimes to get the ACK response, which resulted in unneeded retransmissions. Similar can be observed in logs from customers. In my case, increasing it to 15s, decreased the number of retransmissions heaviliy.

Next, `CONFIG_LWM2M_SECONDS_TO_UPDATE_EARLY` (not yet present in NCS) allows to specify how long before the actual timeout should the Registration Update be sent. So far it was fixed to 6 seconds, which was way to low for NB-IoT network, which resulted in often re-registrations. I set it to 30s to allow for at least one retransmission at CoAP level.

Finally, increasing `CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME` is a consequence of the above.